src/cobra/core/gene.py

```diff
@@ -559,8 +655,99 @@ def _repr_html__(self):
             gpr=format_long_string(self.to_string(), 100)
         )
 
-    # def as_symbolic(self):
-    #     # ...
+    def as_symbolic(
+        self,
+        names: dict = None,
+    ) -> Union[sp_Or, sp_And, Symbol]:
+        """Convert compiled ast to sympy expression.
+
+        Parameters
+        ----------
+        self : GPR
+           compiled ast Module describing GPR
+        names: dict
+           dictionary of gene ids to gene names. If this is empty,
+           returns sympy expression using gene ids
+
+        Returns
+        ------
+        Symbol or BooleanFunction
+            SYMPY expression (Symbol or And or Or). Symbol("") if the GPR is empty
+
+        Notes
+        -----
+        Calls _symbolic_gpr()
+        """
+        # noinspection PyTypeChecker
+        if names:
+            GPRGene_dict = {gid: Symbol(names[gid]) for gid in self.genes}
+        else:
+            GPRGene_dict = None
+        return self._symbolic_gpr(self, GPRGene_dict=GPRGene_dict)
+
+    def _symbolic_gpr(
```

Comment:
That's the one I would prefer. Should be the fastest as well since it directly converts from the pre-parsed tree.